include/zim/writer/item.h

```diff
@@ -61,6 +61,55 @@ namespace zim
         virtual ~Item() = default;
     };
 
+    class BasicItem : public Item
+    {
+      public:
+        BasicItem(const std::string& path, const std::string& mimetype, const std::string title)
+          : path(path),
+            mimetype(mimetype),
+            title(title)
+        {}
+
+        std::string getPath() const { return path; }
+        std::string getTitle() const { return title; }
+        std::string getMimeType() const { return mimetype; }
+
+      protected:
+        std::string path;
+        std::string mimetype;
+        std::string title;
+    };
```

Comment:
I would change this into `struct ItemDescriptor` that is NOT derived from `Item`, and then derive both `StringItem` and `FileItem` from both `Item` and `ItemDescriptor`. The constructor signatures will then be shortened to

```
StringItem(const ItemDescriptor& itemDesc, const std::string& content);
FileItem(const ItemDescriptor& itemDesc, const std::string& filePath);
```